spark/src/test/scala/io/delta/sharing/spark/filters/OpConverterSuite.scala

```diff
@@ -275,4 +275,32 @@ class OpConverterSuite extends SparkFunSuite {
     assert(op.evalExpectBoolean(EvalContext(Map("userId" -> "23"))) == true)
     assert(op.evalExpectBoolean(EvalContext(Map("userId" -> "24"))) == false)
   }
+
+  test("In test") {
+    val sqlColumn = SqlAttributeReference("userId", SqlIntegerType)()
+    val sqlLiterals = Seq(
+      SqlLiteral(23, SqlIntegerType),
+      SqlLiteral(24, SqlIntegerType),
+      SqlLiteral(25, SqlIntegerType),
+      SqlLiteral(26, SqlIntegerType)
+    )
+    val sqlIn = SqlIn(sqlColumn, sqlLiterals)
+    val op = OpConverter.convert(Seq(sqlIn)).get
+    op.validate()
+    assert(op.evalExpectBoolean(EvalContext(Map("userId" -> "22"))) == false)
+    assert(op.evalExpectBoolean(EvalContext(Map("userId" -> "23"))) == true)
+    assert(op.evalExpectBoolean(EvalContext(Map("userId" -> "24"))) == true)
+    assert(op.evalExpectBoolean(EvalContext(Map("userId" -> "25"))) == true)
+    assert(op.evalExpectBoolean(EvalContext(Map("userId" -> "26"))) == true)
+    assert(op.evalExpectBoolean(EvalContext(Map("userId" -> "27"))) == false)
+
+    var sqlLiterals2: Seq[SqlExpression] = Seq.empty
```

Comment:
Can you add a test case where the list is empty and make sure the op is evaluated to `false`?

Reply:
Spark will never send us an empty IN.
In fact, if you specify one entry in the IN operator in sql, spark converts that into an equality check in the expression tree it sends us.

However, I decided to handle both cases.
We throw an error if there are no predicates, and convert it to an equality if there is 1.